spec.md

```diff
@@ -5,4599 +5,421 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
+to such a request MAY return fewer than `<int>` results, but only when the total number of tags attached to the repository
+is less than `<int>`. Otherwise, the response MUST include `<int>` results. Without the `last` query parameter (described
+next), the list returned will start at the beginning of the list and include `<int>` results. As above, the tags MUST be
+in lexical order.
 
-The error codes that MAY be included in the response body are enumerated below:
+The `last` query parameter provides further means for limiting the number of tags. It is used exclusively in combination with the
+`n` parameter:
+[8b](#Endpoints) `/v2/<name>/tags/list?n=<int>&last=<tagname>`
 
-| Code           | Message                 | Description                                                                                                                                                            |
-|----------------|-------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
-| `UNAUTHORIZED` | authentication required | The access controller was unable to authenticate the client. Often this will be accompanied by a Www-Authenticate HTTP response header indicating how to authenticate. |
+`<name>` is the namespace of the repository, `<int>` is the number of tags requested, and `<tagname>` is the *value* of
+the last tag. `<tagname>` MUST NOT be a numerical index, but rather it MUST be a proper tag. A request of this sort will return
+up to `<int>` tags, beginning non-inclusively with `<tagname>`. That is to say, `<tagname>` will not be included in the
+results, but up to `<int>` tags *after* `<tagname>` will be returned. The tags MUST be in lexical order.
 
-###### On Failure: Too Many Requests
+#### Content Management
+Content management refers to the deletion of blobs, tags, and manifests. Registries MAY implement deletion or they MAY
+disable it. Similarly, a registry MAY implement tag deletion, while others MAY allow deletion only by manifest.
 
-```HTTP
-429 Too Many Requests
-Content-Length: <length>
-Content-Type: application/json
-
-{
-    "errors": [
-        {
-            "code": "<error code>",
-            "message": "<error message>",
-            "detail": ...
-        },
-        ...
-    ]
-}
-```
-
-The client made too many requests within a time interval.
-
-The following headers will be returned on the response:
-
-| Name             | Description                       |
-|------------------|-----------------------------------|
-| `Content-Length` | Length of the JSON response body. |
+##### Deleting tags
+`<name>` is the namespace of the repository, and `<tag>` is the name of the tag to be deleted. Upon success, the registry
+MUST respond with a `202 Accepted` code. If tag deletion is disabled, the registry MUST respond with either a
+`400 Bad Request` or a `405 Method Not Allowed`.
```

Comment:
Does the spec want to take a stand on auto-deletion of a manifest when the last tag to it is removed? In particular, should that be disallowed? Else clients may have to try and guess what a repo will do...

Reply:
The spec doesn't take a stand on this, and registries implement a fairly wide range of garbage-collecting behavior.  It is worth opening an issue to discuss, possibly for inclusion in a post-1.0 spec.

Reply:
I opened the issue here: #180